bin/common

```diff
@@ -0,0 +1,138 @@
+unset verbose quiet cygwin toolcp colors saved_stty CDPATH
+
+function find_jar {
+    # Usage:
+    #   find_jar path/to/location file.jar
+    local artifact="$1/$2"
+
+    if [ ! -f "$artifact" ]; then
+        artifact=$(find "$HOME/.coursier/cache" -iname "$2")
+    fi
+
+    echo "$artifact"
+}
+
+# Autodetecting the scala-library location, in case it wasn't provided by an environment variable
+if [ "$SCALA_LIBRARY_JAR" == "" ]; then
+    SCALA_LIBRARY_JAR=$(find_jar "$HOME/.ivy2/cache/org.scala-lang/scala-library/jars" "scala-library-$SCALA_VERSION.jar")
+fi
+
+if [ "$SCALA_REFLECT_JAR" == "" ]; then
+    SCALA_REFLECT_JAR=$(find_jar "$HOME/.ivy2/cache/org.scala-lang/scala-reflect/jars" "scala-reflect-$SCALA_VERSION.jar")
+fi
+
+if [ "$SCALA_COMPILER_JAR" == "" ]; then
+    SCALA_COMPILER_JAR=$(find_jar "$HOME/.ivy2/cache/me.d-d/scala-compiler/jars" "scala-compiler-$SCALA_COMPILER_VERSION.jar")
+fi
+
+if [ "$JLINE_JAR" == "" ]; then
+    JLINE_JAR=$(find_jar "$HOME/.ivy2/cache/jline/jline/jars" "jline-$JLINE_VERSION.jar")
+fi
+
+if [ "$SBT_INTERFACE_JAR" == "" ]; then
+    SBT_INTERFACE_JAR=$(find_jar "$HOME/.ivy2/cache/org.scala-sbt/interface/jars" "interface-$SBT_VERSION.jar")
+fi
+
+function build_jar {
+    # Usage:
+    #   build_jar package path/to/jar/dir ['/some/sed/command']
+    #
+    #   Last arg is optional
+    cd $DOTTY_ROOT >& /dev/null
+    local build_output=$(sbt "$1")
+    local jar=$(echo $build_output | sed -n 's/.*Packaging //g; s/ \.\.\..*//g; /^\/.*/p')
+
+    local sedjar="$3"
+    if [ "$sedjar" == "" ]; then
+        sedjar="/.*\.jar/p"
+    fi
+
+    if [ "$jar" == "" ]; then
+        # Didn't build a jar - could've run sbt by oneself, get latest jar in target:
+        jar="$DOTTY_ROOT/$2/$(ls -1t "$2" | sed -n "$sedjar" | awk 'NR==1')"
+    fi
+
+    cd - >& /dev/null
+
+    echo $jar
+}
+
+function update_packages {
+    echo "$INTERFACES_JAR" >  $DOTTY_ROOT/.packages
+    echo "$MAIN_JAR"       >> $DOTTY_ROOT/.packages
+    echo "$TEST_JAR"       >> $DOTTY_ROOT/.packages
+}
+
+function build_all {
+    echo "The script is going to build the required jar files"
+
+    printf "Building dotty-interfaces..."
+    INTERFACES_JAR=$(build_jar dotty-interfaces/package interfaces/target)
+    printf "done\n"
+
+    printf "Building dotty..."
+    MAIN_JAR=$(build_jar package target/scala-2.11)
+    printf "done\n"
+
+    printf "Building tests..."
+    TEST_JAR=$(build_jar test:package target/scala-2.11 '/dotty.*-tests\.jar/p')
+    printf "done\n"
+
+    update_packages
+}
+
+# Check if .packages file does not exist - if so assume old build and rebuild all
+if  [ ! -f "$DOTTY_ROOT/.packages" ]; then
+    build_all
+else
+    IFS=$'\r\n' GLOBIGNORE='*' command eval 'JARS=($(cat $DOTTY_ROOT/.packages))'
+
+    if [ "${#JARS[@]}" == "3" ]; then
+        INTERFACES_JAR="${JARS[0]}"
+        MAIN_JAR="${JARS[1]}"
+        TEST_JAR="${JARS[2]}"
+    else
+        echo "Corrupted .packages file"
```

Comment:
Instead of `corrupted` I'd report `failed to parse`, as corrupted seems to aggressive. I'd assume file system failure if I see such error message.